### PR TITLE
Add benefits_claims_id to form_submissions table

### PR DESCRIPTION
**Pull request by Thrillberg**
## Summary
This PR resuscitates [this old PR](https://github.com/department-of-veterans-affairs/vets-api/pull/17500) to extend our `FormSubmission` model to handle not only forms that go to Benefits Intake API but also the Benefits Claims API. For our particular use case we'll want to save some other data from the response but thankfully the `FormSubmissionAttempt` has a jsonb field, `response`. So we should be all set on that front!

### CODE REVIEW

**Review by stevenjcumming — APPROVED** [review]

File: db/migrate/20241011122123_add_benefits_claims_id_to_form_submissions.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddBenefitsClaimsIdToFormSubmissions < ActiveRecord::Migration[7.1]
+  def change
+    add_column :form_submissions, :benefits_claims_id, :string
```
Comment (wayne-weibel) [inline db/migrate/20241011122123_add_benefits_claims_id_to_form_submissions.rb:3]: instead use `benefits_claims_uuid`?

**Review by wayne-weibel — CHANGES_REQUESTED** [review]
Will also want to update the `BenefitsIntakeStatusJob` and`BenefitsIntakeRemediationStatusJob` to filter out the rows which do NOT have `benefits_intake_uuid`

File: db/migrate/20241011122123_add_benefits_claims_id_to_form_submissions.rb (hunk 1 above)
Comment (Thrillberg) [reply db/migrate/20241011122123_add_benefits_claims_id_to_form_submissions.rb:3]: I thought about this but the Benefits Claims API seems to return an incrementing integer id, not a UUID. The Benefits Intake API returns a UUIDv4.

**Comment by Thrillberg** [comment]
> Will also want to update the `BenefitsIntakeStatusJob` and`BenefitsIntakeRemediationStatusJob` to filter out the rows which do NOT have `benefits_intake_uuid`

I think `vets-api` prefers to have db commits be isolated like this. But yeah once we start writing to this field we'll have to be careful to update the jobs.

**Comment by wayne-weibel** [comment]
https://dsva.slack.com/archives/C053U7BUT27/p1729020835555299

**Comment by Thrillberg** [comment]
Sounds like we don't need this now anyway, closing.

**Pull request closed without merging**